src/sitemap.rs

```diff
@@ -489,6 +503,33 @@ impl SitemapElement {
             SitemapElement::TocItem(s) => Some(s.id.clone()),
         }
     }
+
+    // If url contains path parameters so it will set those parameters
+    // /person/<string:username>/<integer:age>
+    // In that case it will parse and set parameters `username` and `age`
+    pub(crate) fn set_path_params(&mut self, url: &str) {
+        lazy_static::lazy_static! {
+            static ref PATH_PARAMS: regex::Regex = regex::Regex::new(r"<\s*([a-z]\w+)\s*:\s*([a-z|A-Z|0-9|_]\w+)\s*>")
+            .expect("PATH_PARAMS: Regex is wrong");
+        }
+        let params = PATH_PARAMS
+            .captures_iter(url)
+            .into_iter()
+            .map(|params| (params[1].to_string(), params[2].to_string()))
+            .collect::<Vec<_>>();
```

Comment:
Extract this logic out in a pure function which takes url and returns Vec<(String, String)>. So it can be tested independently, without creating `Self` etc.